grpc.go

```diff
@@ -536,6 +541,91 @@ func (a *agentGRPC) WaitProcess(ctx context.Context, req *pb.WaitProcessRequest)
 	}, nil
 }
 
+func getPIDIndex(title string) int {
+	// looking for PID field in ps title
+	fields := strings.Fields(title)
+	for i, f := range fields {
+		if f == "PID" {
+			return i
+		}
+	}
+	return -1
+}
+
+func (a *agentGRPC) ListProcesses(ctx context.Context, req *pb.ListProcessesRequest) (*pb.ListProcessesResponse, error) {
+	resp := &pb.ListProcessesResponse{}
+
+	c, err := a.sandbox.getContainer(req.ContainerId)
+	if err != nil {
+		return resp, err
+	}
+
+	pids, err := c.container.Processes()
+	if err != nil {
+		return resp, err
+	}
+
+	switch req.Format {
+	case "table":
+	case "json":
+		resp.ProcessList, err = json.Marshal(pids)
+		return resp, err
+	default:
+		return resp, fmt.Errorf("invalid format option")
+	}
+
+	psArgs := req.Args
+	if len(psArgs) == 0 {
+		psArgs = []string{"-ef"}
+	}
+
+	cmd := exec.Command("ps", psArgs...)
+	output, err := cmd.CombinedOutput()
```

Comment:
Have you tried to run this intensively ? I ask because I am afraid this command might error sometimes, due to the subreaper. Let's say you run your `ps` process from the agent, the subreaper might be the one reaping the process when it terminates, meaning the `cmd.Process.Wait()` called from `cmd.CombinedOutput()` might return an error because it didn't properly got the exit code from the command.
@laijs @bergwolf any input on that ?

Reply:
you're right, fixed